mapdataprefetcher/app/src/main/java/com/here/android/example/map/data/prefetcher/MapFragmentView.java

```diff
@@ -0,0 +1,440 @@
+                if ((dataSizeKB / 1000) > CACHE_SIZE) {
+                    Toast.makeText(m_activity, m_activity.getText(R.string.error_map_size),
+                            Toast.LENGTH_SHORT).show();
+                } else {
+                    if (m_mapType == MapType.ROAD_RADIUS) {
+                        Toast.makeText(m_activity, m_activity.getText(R.string.prefetching_map),
+                                Toast.LENGTH_SHORT).show();
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void onStatus(int requestId, PrefetchStatus status) {
+            super.onStatus(requestId, status);
+            if (status == PrefetchStatus.PREFETCH_SUCCESS) {
+                if (m_mapType == MapType.MAP_BOUNDING_BOX) {
+                    showProgressBar(false);
+                    changeMapState(MapState.CALCULATE_ROUTE);
+
+                    m_map.zoomTo(m_geoBoundingBox, Map.Animation.NONE, 0);
+                } else {
+                    showProgressBar(false);
+                    changeMapState(MapState.SIMULATE_NAVIGATION);
+
+                    m_map.zoomTo(m_geoBoundingBox, m_map.getWidth() - 200, m_map.getHeight() - 200,
+                            Map.Animation.NONE, 0);
+                }
+
+                MapEngine.setOnline(false);
+            }
+        }
+    }
+
+    private void changeMapState(MapState mapState) {
+        this.m_mapState = mapState;
+        m_btnChangeMapState.setVisibility(View.VISIBLE);
+
+        switch (mapState) {
+            case DEFAULT:
+                m_btnChangeMapState.setVisibility(View.GONE);
+                break;
+            case PREFETCH_MAP:
+                if (m_mapType == MapType.MAP_BOUNDING_BOX) {
+                    m_btnChangeMapState.setText(R.string.prefetch_geo_box);
+                } else {
+                    m_btnChangeMapState.setText(R.string.prefetch_route);
+                }
+                break;
+            case CALCULATE_ROUTE:
+                m_btnChangeMapState.setText(R.string.calculate_route);
+                break;
+            case SIMULATE_NAVIGATION:
+                m_btnChangeMapState.setText(R.string.simulate_navigation);
+                break;
+            case STOP_NAVIGATION:
+                m_btnChangeMapState.setText(R.string.stop_navigation);
+                break;
+        }
+    }
+
+    private void showRoute() {
+        m_startNavPoint = new GeoCoordinate(55.93176, -3.22734);
+        m_endNavPoint = new GeoCoordinate(55.98084, -3.17778);
+
+        calculateRoute(m_startNavPoint, m_endNavPoint);
+
+        List<GeoCoordinate> geoCoordinateList = new ArrayList<>();
+        geoCoordinateList.add(m_startNavPoint);
+        geoCoordinateList.add(m_endNavPoint);
+
+        m_geoBoundingBox = GeoBoundingBox
+                .getBoundingBoxContainingGeoCoordinates(geoCoordinateList);
+        m_map.zoomTo(m_geoBoundingBox, m_map.getWidth() - 200, m_map.getHeight() - 200,
```

Comment:
Do not use magic values, like 200.  Name them, move at the top of the file and make  `final static ...`. (https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad)